pipelines/enterprise-contract.yaml

```
@@ -59,16 +59,41 @@ spec:
       type: string
       description: The name of the key in the ConfigMap that contains the CA bundle data.
       default: "ca-bundle.crt"
+    - name: REDUCE_SNAPSHOT
+      type: string
+      description: Reduce the Snapshot to only the component whose build caused the Snapshot to be created
+      default: "false"
+    - name: PIPELINERUN_ID
+      type: string
+      description: PipelineRun ID
+      default: "$(context.pipelineRun.name)"
   results:
     - name: TEST_OUTPUT
       value: "$(tasks.verify.results.TEST_OUTPUT)"
   tasks:
+    - name: reduce-snapshot
+      params:
+        - name: SNAPSHOT
+          value: "$(params.SNAPSHOT)"
+        - name: REDUCE_SNAPSHOT
+          value: "$(params.REDUCE_SNAPSHOT)"
+        - name: PIPELINERUN_ID
+          value: "$(params.PIPELINERUN_ID)"
+      taskRef:
+        resolver: "git"
+        params:
+          - name: url
+            value: https://github.com/konflux-ci/build-definitions
+          - name: revision
+            value: main
+          - name: pathInRepo
+            value: task/reduce-snapshot/0.1/reduce-snapshot.yaml
```

I think we should do what is done in other pipelines here, the `hack/build-and-push.sh` should replace this with the newly built Task bundle and configure the bundle resolver
```suggestion
      taskRef:
        name: reduce-snapshot
        version: "0.1"
```

---

changed. thanks!

---

actually, this does not work since the EC pipeline is executed via a Git resolver from the ITS ... if I use a git resolver, it works.